### PR TITLE
Fix some type errors.

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
mypy was reporting a few new type errors from recent PRs. These address them.

**Testing steps**
```
roller-mbp parlai typefixes $ git commit -am "Fix some type errors."
[typefixes ad2526ee] Fix some type errors.
 2 files changed, 13 insertions(+), 6 deletions(-)
roller-mbp parlai typefixes $ mypy
Success: no issues found in 784 source files
```

### CODE REVIEW

File: parlai/agents/transformer/modules.py [hunk 1]
```diff
@@ -1077,10 +1079,15 @@ def prepare_head(tensor):
         if key is None and value is None:
             # self attention
             key = value = query
+            _, _key_len, dim = query.size()
```
Comment (EricMichaelSmith) [inline parlai/agents/transformer/modules.py]: hmm, is there a need to define these vars both here and in line 1090 below?

File: parlai/agents/transformer/modules.py [hunk 2]
```diff
@@ -1025,13 +1025,15 @@ def __init__(self, n_heads, dim, dropout=0):
...
 
-    def forward(
+    def forward(  # type: ignore
+        # TODO: remove type ignore with pytorch 1.5:
+        # https://github.com/pytorch/pytorch/pull/31057
         self,
         query: torch.Tensor,
-        key: torch.Tensor = None,
-        value: torch.Tensor = None,
+        key: Optional[torch.Tensor] = None,
```
Comment (EricMichaelSmith) [inline parlai/agents/transformer/modules.py]: Optional[] is great!

**Comment by klshuster** [comment]
looks good to me, however your new runtime error is making a lot of our tests fail...

File: parlai/agents/transformer/modules.py (hunk 1 above)
Comment (stephenroller) [reply parlai/agents/transformer/modules.py]: Yeah it was to make the typechecker happy.

**Pull request merged**